prow/cmd/mkpod/main.go

```diff
@@ -98,3 +138,47 @@ func main() {
 	}
 	fmt.Println(string(podYAML))
 }
+
+func makeLocalPod(pj prowapi.ProwJob, buildID, outDir string) (*v1.Pod, error) {
+	pod, err := decorate.ProwJobToPodLocal(pj, buildID, outDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prompt for emptyDir or hostPath replacements for all volume sources besides those two.
+	volsToFix := nonLocalVolumes(pod.Spec.Volumes)
+	if len(volsToFix) > 0 {
+		prompt := `For each of the following volumes specify one of:
+ - 'empty' to use an emptyDir;
+ - a path on the host to use hostPath;
+ - '' (nothing) to use the existing volume source and assume it is available in the cluster`
+		fmt.Fprintln(os.Stderr, prompt)
+		for _, vol := range volsToFix {
+			fmt.Fprintf(os.Stderr, "Volume %q: ", vol.Name)
+
+			var choice string
+			fmt.Scanln(&choice)
```

Comment:
Often it's nice to allow this to be flag-driven so it can be scripted, if for instance the UX to deliver to devs is this tool wrapped with some more context (maybe create tempdirs for you etc)